integration/storagetest/logtests.go

```diff
@@ -535,3 +518,68 @@ func ensureAllLeavesDistinct(t *testing.T, leaves []*trillian.LogLeaf) {
 		set[k] = true
 	}
 }
+
+func (*logTests) TestAddSequencedLeavesAndDequeueLeaves(ctx context.Context, t *testing.T, s storage.LogStorage, as storage.AdminStorage) {
+	tree := mustCreateTree(ctx, t, as, storageto.PreorderedLogTree)
+	mustSignAndStoreLogRoot(ctx, t, s, tree, &types.LogRootV1{})
+
+	leaves := createTestLeaves(3, 0)
+	leaves[0].ExtraData = nil
+
+	now := time.Now()
+	res, err := s.AddSequencedLeaves(ctx, tree, leaves, now)
+	if err != nil {
+		t.Errorf("AddSequencedLeaves(_, %v, %+v, %v): %v", tree.TreeId, leaves, now, err)
+	}
+
+	if got, want := len(res), len(leaves); got != want {
+		t.Fatalf("AddSequencedLeaves returned %v results, want %v", got, want)
+	}
+	for i, r := range res {
+		if got, want := status.FromProto(r.Status).Code(), codes.OK; got != want {
+			t.Errorf("res[%d]=%v, want %v", i, got, want)
+		}
+	}
+
+	qts, _ := ptypes.TimestampProto(now)
+	its, _ := ptypes.TimestampProto(time.Unix(0, 0))
+
+	partial := make([]*trillian.LogLeaf, 0, len(leaves))
+	for _, leaf := range leaves {
+		partial = append(partial, &trillian.LogLeaf{
+			LeafIdentityHash:   leaf.LeafIdentityHash,
+			MerkleLeafHash:     leaf.MerkleLeafHash,
+			LeafValue:          leaf.LeafValue,
+			ExtraData:          leaf.ExtraData,
+			LeafIndex:          leaf.LeafIndex,
+			QueueTimestamp:     qts,
+			IntegrateTimestamp: its,
+		})
+	}
+
+	// Check that the first sequenced entry is returned.
+	dequeued, err := dequeueLeavesInTx(ctx, s, tree, now, 1)
+	if err != nil {
+		t.Fatalf("dequeueLeaves(): %v", err)
+	}
+	if want := partial[:1]; !cmp.Equal(dequeued, want, cmp.Comparer(proto.Equal)) {
```

Comment:
I wonder if you could reduce this to 1 call:
```golang
if diff := cmp.Diff(...); diff != "" {
	t.Errorf(..., diff)
}
```
Same 15 lines below.

Reply:
done